activation/nipost.go

```diff
@@ -233,15 +219,30 @@ func (nb *NIPostBuilder) BuildNIPost(ctx context.Context, challenge *types.PoetC
 }
 
 // Submit the challenge to a single PoET.
-func (nb *NIPostBuilder) submitPoetChallenge(ctx context.Context, poet PoetProvingServiceClient, challenge []byte, signature types.EdSignature) (*types.PoetRequest, error) {
+func (nb *NIPostBuilder) submitPoetChallenge(ctx context.Context, poet PoetProvingServiceClient, prefix, challenge []byte, signature types.EdSignature, nodeID types.NodeID) (*types.PoetRequest, error) {
 	poetServiceID, err := poet.PoetServiceID(ctx)
 	if err != nil {
 		return nil, &PoetSvcUnstableError{msg: "failed to get PoET service ID", source: err}
 	}
 	logger := nb.log.WithContext(ctx).WithFields(log.String("poet_id", hex.EncodeToString(poetServiceID.ServiceID)))
+
+	logger.Debug("Querying for poet PoW parameters")
+	powParams, err := poet.PowParams(ctx)
+	if err != nil {
+		return nil, &PoetSvcUnstableError{msg: "failed to get PoW params", source: err}
+	}
+	logger.Debug("Doing Pow with params: %v", powParams)
+	nonce, err := shared.FindSubmitPowNonce(ctx, powParams.Challenge, challenge, nodeID.Bytes(), powParams.Difficulty)
```

Comment:
please add a metric to track poet_pow_latency and maybe a warning if this takes longer then some reasonable upper bounder (1 minute is ok?)